src/Sylius/Behat/Context/Ui/Shop/ProductContext.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Behat\Context\Ui\Shop;
+
+use Behat\Behat\Context\Context;
+use Sylius\Behat\Page\Shop\Product\ShowPageInterface;
+use Sylius\Component\Core\Model\ProductInterface;
+use Webmozart\Assert\Assert;
+
+/**
+ * @author Kamil Kokot <kamil.kokot@lakion.com>
+ * @author Magdalena Banasiak <magdalena.banasiak@lakion.com>
+ * @author Anna Walasek <anna.walasek@lakion.com>
+ */
+final class ProductContext implements Context
+{
+    /**
+     * @var ShowPageInterface
+     */
+    private $showPage;
+
+    /**
+     * @param ShowPageInterface $showPage
+     */
+    public function __construct(ShowPageInterface $showPage)
+    {
+        $this->showPage = $showPage;
+    }
+
+    /**
+     * @Then I should be able to access product :product
+     */
+    public function iShouldBeAbleToAccessProduct(ProductInterface $product)
+    {
+        $this->showPage->tryToOpen(['product' => $product]);
+
+        expect($this->showPage->isOpen(['product' => $product]))->toBe(true);
+    }
+
+    /**
+     * @Then I should not be able to access product :product
+     */
+    public function iShouldNotBeAbleToAccessProduct(ProductInterface $product)
+    {
+        $this->showPage->tryToOpen(['product' => $product]);
+
+        expect($this->showPage->isOpen(['product' => $product]))->toBe(false);
+    }
+
+    /**
+     * @When /^I select (this product) from list$/
+     */
+    public function iSelectThisProductFromList(ProductInterface $product)
+    {
+        $this->showPage->open(['slug' => $product->getSlug()]);
+    }
+
+    /**
+     * @Then /^I should see (this product) detailed page$/
+     */
+    public function iShouldSeeProductDetailedPage(ProductInterface $product)
+    {
+        expect($this->showPage->isOpen(['slug' => $product->getSlug()]))->toBe(true);
+    }
+
+    /**
+     * @Given I should see the product name :name
+     */
+    public function iShouldSeeProductName($name)
+    {
+        Assert::same(
+            $name,
+            $this->showPage->getName(),
+            sprintf('Product should have name %s, but it has %s', $name, $this->showPage->getName())
```

Comment:
No need for `sprintf` the string `'Product should have name %2$s, but it has %s'` will be enough.